### ISSUE TITLE
NullReferenceException when resizing an array of "ClipState.Transition" 

### ISSUE THREAD

**Issue by DaPaKnat** [KybernetikGames/animancer#38]
# Environment

- Animancer Version Number: 4.4
- Animancer Pro or Lite: Pro
- Unity Version: 2019.3.7f1
- Platform: Windows 10

# Description

Hi, First of all, really nice work there, I've been looking for a more flexible animation system for a while and I bought it for a school project (and many more in the future). However, I encountered a problem resizing a serialized field of "ClipState.Transition" array. The array is initialized with 0 elements and when I resize the array, I get a NullReferenceException on "GetValue" in this part of the code :

`Instance.Property = transitionProperty;`
`Instance.Transition = transitionProperty.GetValue<ITransitionDetailed>();` `Instance.MaximumDuration = Instance.Transition.MaximumDuration;`

I've also tried to resize an existing working array in one of your example (SequenceCoroutine) and it shows the same error.

# Reproduction

1. Go to 01 Basics/03 Sequence Coroutine.
2. Select the "LowPolyMan"
3. Try to resize the "Animations" array in the "Basics - Sequence Coroutine" script and it should pop an error.

**Comment by KybernetikGames** [comment]
I'm glad you like Animancer and thanks for the bug report.

It looks like Unity does something differently when you have a custom Inspector for your script compared to having none so I never saw the issue because my Inspector Gadgets Pro plugin applies its custom Inspector to every script.

I seem to have fixed it with a couple of changes:

- Change that 3rd line to `Instance.MaximumDuration = Instance.Transition != null ? Instance.Transition.MaximumDuration : 0;`
- Find all references to that `TransitionContext.Get` method (one in TransitionDrawer.cs and one in ManualMixerState.cs) and put a null check right after them:

```
using (TransitionContext.Get(this, property))
{
    if (Context.Transition == null)
        return;
```

**Comment by DaPaKnat** [comment]
Thank you very much, it worked like a charm!

**Comment by KybernetikGames** [comment]
Great. I'll keep the issue open so it's visible until I release the next version.

**Comment by KybernetikGames** [comment]
This issue is fixed in [Animancer v5.0 Alpha 1](https://kybernetikgames.github.io/animancer/docs/changes/animancer-v5-0).
